ray-operator/controllers/ray/metrics/ray_cluster_metrics.go

```diff
@@ -6,17 +6,37 @@ import (
 
 type RayClusterMetricCollector struct {
 	// Metrics
+	rayClusterProvisionedDuration *prometheus.GaugeVec
 }
 
 func NewRayClusterMetricCollector() *RayClusterMetricCollector {
-	collector := &RayClusterMetricCollector{}
+	collector := &RayClusterMetricCollector{
+		rayClusterProvisionedDuration: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "kuberay_cluster_provisioned_duration_seconds",
+				Help: "The time, in seconds, when a RayCluster's `RayClusterProvisioned` status transitions from false (or unset) to true",
+			},
+			[]string{"name", "namespace"},
+		),
+	}
 	return collector
 }
 
 // Describe implements prometheus.Collector interface Describe method.
-func (c *RayClusterMetricCollector) Describe(_ chan<- *prometheus.Desc) {
+func (c *RayClusterMetricCollector) Describe(ch chan<- *prometheus.Desc) {
+	c.rayClusterProvisionedDuration.Describe(ch)
 }
 
 // Collect implements prometheus.Collector interface Collect method.
-func (c *RayClusterMetricCollector) Collect(_ chan<- prometheus.Metric) {
+func (c *RayClusterMetricCollector) Collect(ch chan<- prometheus.Metric) {
+	c.rayClusterProvisionedDuration.Collect(ch)
+}
+
+// ObserveRayClusterProvisionedDuration observes the duration of RayCluster from creation to provisioned
+func (c *RayClusterMetricCollector) ObserveRayClusterProvisionedDuration(name, namespace string, duration float64) {
+	c.rayClusterProvisionedDuration.WithLabelValues(name, namespace).Set(duration)
+}
+
+func (c *RayClusterMetricCollector) DeleteRayClusterProvisionedDuration(name, namespace string) {
+	c.rayClusterProvisionedDuration.DeleteLabelValues(name, namespace)
```

Comment:
1. Is it a common practice to call `DeleteLabelValues`?
2. What is the impact on users and KubeRay after calling `DeleteLabelValues`? For the latter, perhaps we can save some memory.
3. What will happen if we create a new RayCluster CR with the same name / namespace?

<img width="626" alt="image" src="https://github.com/user-attachments/assets/b80cf44e-e0cc-4e97-acda-74c8e71e8e5b" />